pkg/pcrbruteforcer/reproduce_event_log_test.go

```diff
@@ -22,70 +22,96 @@ func unhex(fataler fataler, h string) []byte {
+					Digest:   unhex(fataler, "0000000000000000000000000000000000000000"),
 				},
 			},
 			{
 				PCRIndex: 0,
 				Type:     tpmeventlog.EV_S_CRTM_CONTENTS,
 				Digest: &tpmeventlog.Digest{
 					HashAlgo: 4,
-					Digest:   unhex(t, "527C9A38B2F45FBF89C382547E0A0812722A47D3"),
+					Digest:   unhex(fataler, "527C9A38B2F45FBF89C382547E0A0812722A47D3"),
 				},
 			},
 			{
 				PCRIndex: 0,
 				Type:     tpmeventlog.EV_S_CRTM_VERSION,
 				Digest: &tpmeventlog.Digest{
 					HashAlgo: 4,
-					Digest:   unhex(t, "C14F556E35C9BB45F189B03F383A6A3E31256681"),
+					Digest:   unhex(fataler, "C14F556E35C9BB45F189B03F383A6A3E31256681"),
 				},
 			},
 			{
 				PCRIndex: 0,
 				Type:     tpmeventlog.EV_POST_CODE,
 				Digest: &tpmeventlog.Digest{
 					HashAlgo: 4,
-					Digest:   unhex(t, "4C9836F73CC42ADBECE7D565B783E618B4A75C22"),
+					Digest:   unhex(fataler, "4C9836F73CC42ADBECE7D565B783E618B4A75C22"),
 				},
 			},
 			{
 				PCRIndex: 0,
 				Type:     tpmeventlog.EV_SEPARATOR,
 				Digest: &tpmeventlog.Digest{
 					HashAlgo: 4,
-					Digest:   unhex(t, "9069CA78E7450A285173431B3E52C5C25299E473"),
+					Digest:   unhex(fataler, "9069CA78E7450A285173431B3E52C5C25299E473"),
 				},
 			},
 		},
 	}
+}
+
+func TestReproduceEventLog(t *testing.T) {
+	firmwareImage := firmware.FakeIntelFirmware
+
+	firmware, err := uefi.ParseUEFIFirmwareBytes(firmwareImage)
+	require.NoError(t, err)
 
-	succeeded, acmPolicyStatus, _ := ReproduceEventLog(eventLog, measurements, firmwareImage)
+	measureOptions := []pcr.MeasureOption{
+		pcr.SetFlow(pcr.FlowIntelCBnT0T),
+		pcr.SetIBBHashDigest(tpm2.AlgSHA1),
+		pcr.SetRegisters(registers.Registers{
+			registers.ParseACMPolicyStatusRegister(0x0000000200108682),
+		}),
+	}
+	measurements, _, debugInfo, err := pcr.GetMeasurements(firmware, 0, measureOptions...)
+	require.NoError(t, err, fmt.Sprintf("debugInfo: '%v'", debugInfo))
+
+	eventLog := getTPMEventLog(t)
+
+	succeeded, acmPolicyStatus, _ := ReproduceEventLog(eventLog, tpmeventlog.TPMAlgorithmSHA1, measurements, firmwareImage, DefaultSettingsReproduceEventLog())
 	require.True(t, succeeded)
-	require.NotNil(t, acmPolicyStatus)
 	require.Equal(t, uint64(0x0000000200108681), acmPolicyStatus.Raw())
 }
+
+func BenchmarkReproduceEventLog(b *testing.B) {
+	firmware := getFirmware(b)
+
+	measureOptions := []pcr.MeasureOption{
+		pcr.SetFlow(pcr.FlowIntelCBnT0T),
+		pcr.SetIBBHashDigest(tpm2.AlgSHA1),
+		pcr.SetRegisters(registers.Registers{
+			registers.ParseACMPolicyStatusRegister(0x000000020010868A),
+		}),
+	}
+	measurements, _, debugInfo, err := pcr.GetMeasurements(firmware, 0, measureOptions...)
+	require.NoError(b, err, fmt.Sprintf("debugInfo: '%v'", debugInfo))
+
+	eventLog := getTPMEventLog(b)
+
+	for i := 0; i < b.N; i++ {
+		_, _, err := ReproduceEventLog(eventLog, tpmeventlog.TPMAlgorithmSHA1, measurements, firmware.Buf(), DefaultSettingsReproduceEventLog())
+		if err != nil {
+			panic(err)
+		}
```

Comment:
Can you guess? :D

Reply:
Fixed.

Reply:
(fixed in 5cbd9a42c406a6c29b7284088440fc6497c987bb)